generic/tests/ping.py

```diff
@@ -9,8 +9,12 @@
 def _ping_with_params(test, params, dest, interface=None,
                       packet_size=None, interval=None,
                       count=0, session=None, flood=False):
+
+    if (not interval or interval < 0.2):
+        interval = 0.2  # minimal allowed interval for ping without root priveleges
```

Comment:
Nice point Sergey! I didn't notice this till now.

Tried with "ping $ip -i 0.0001 -s 1024" without root privilege and got the output below:
ping: cannot flood; minimal interval allowed for user is 200ms

So the OS does have the limitation of ping.

However, I think it's better to prompt some warning message first and then set interval to 0.2.
You have to let user know there's the limitation instead of automatically covered it.
What do you think?